plugins/device/deviceutils/store-utils.go

```diff
@@ -1,166 +1,146 @@
 package deviceutils
 
 import (
+	"context"
 	"encoding/json"
-	"errors"
-	"strings"
 
 	"github.com/gluster/glusterd2/glusterd2/brick"
 	"github.com/gluster/glusterd2/glusterd2/gdctx"
-	peer "github.com/gluster/glusterd2/glusterd2/peer"
+	"github.com/gluster/glusterd2/glusterd2/store"
 	gderrors "github.com/gluster/glusterd2/pkg/errors"
 	"github.com/gluster/glusterd2/pkg/lvmutils"
 	deviceapi "github.com/gluster/glusterd2/plugins/device/api"
+
+	"github.com/coreos/etcd/clientv3"
+)
+
+const (
+	devicePrefix string = "devices/"
 )
 
 // GetDevices returns devices of specified peer/peers from the store
 // if no peers are specified, it returns devices of all peers
 func GetDevices(peerIds ...string) ([]deviceapi.Info, error) {
-
-	var peers []*peer.Peer
+	var devices []deviceapi.Info
 	var err error
+	var resp *clientv3.GetResponse
+
 	if len(peerIds) > 0 {
 		for _, peerID := range peerIds {
-			var peerInfo *peer.Peer
-			peerInfo, err = peer.GetPeer(peerID)
+			resp, err = store.Get(context.TODO(), devicePrefix+peerID+"/", clientv3.WithPrefix())
 			if err != nil {
 				return nil, err
 			}
-			peers = append(peers, peerInfo)
 		}
 	} else {
-		peers, err = peer.GetPeers()
+		resp, err = store.Get(context.TODO(), devicePrefix, clientv3.WithPrefix())
 		if err != nil {
 			return nil, err
 		}
 	}
-	var devices []deviceapi.Info
-	for _, peerInfo := range peers {
-		deviceInfo, err := GetDevicesFromPeer(peerInfo)
-		if err != nil {
+	for _, kv := range resp.Kvs {
+		var dev deviceapi.Info
+
+		if err = json.Unmarshal(kv.Value, &dev); err != nil {
 			return nil, err
 		}
-		devices = append(devices, deviceInfo...)
+		devices = append(devices, dev)
 	}
 	return devices, nil
 }
 
-// GetDevicesFromPeer returns devices from peer object.
-func GetDevicesFromPeer(peerInfo *peer.Peer) ([]deviceapi.Info, error) {
+// GetDevice returns device of specified peer and device name
+func GetDevice(peerID, deviceName string) (*deviceapi.Info, error) {
+	var device deviceapi.Info
+	var err error
 
-	var deviceInfo []deviceapi.Info
-	if _, exists := peerInfo.Metadata["_devices"]; exists {
-		if err := json.Unmarshal([]byte(peerInfo.Metadata["_devices"]), &deviceInfo); err != nil {
-			return nil, err
-		}
+	resp, err := store.Get(context.TODO(), devicePrefix+peerID+"/"+deviceName)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.Count != 1 {
+		return nil, gderrors.ErrDeviceNotFound
+	}
+
+	if err = json.Unmarshal(resp.Kvs[0].Value, &device); err != nil {
+		return nil, err
 	}
-	return deviceInfo, nil
+
+	return &device, nil
 }
 
 // SetDeviceState sets device state and updates device state in etcd
 func SetDeviceState(peerID, deviceName, deviceState string) error {
-
-	devices, err := GetDevices(peerID)
+	resp, err := store.Get(context.TODO(), devicePrefix+peerID+"/"+deviceName)
```

Comment:
can we call `GetDevice(PeerID,DeviceName)` ? to avoid code duplication?

Reply:
makes sense. will change it